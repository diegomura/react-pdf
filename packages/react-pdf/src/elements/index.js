import Text from './Text';
import View from './View';
import Page from './Page';
import Document from './Document';
import omit from 'lodash/fp/omit';

function createElement(type, props, root) {
  let instance;

  switch (type) {
    case 'ROOT':
      instance = new Document();
      break;
    case 'DOCUMENT':
      /*
       Since the Document instance is being created by the root element,
       all we do when the renderer gets to <Document /> is injecting
       it's props and return the root
      */
      root.setProps(omit('children', props));
      instance = root;
      break;
    case 'PAGE':
      instance = new Page(props, root);
      break;
    case 'TEXT':
      instance = new Text(props, root);
      break;
    case 'VIEW':
      instance = new View(props, root);
      break;
    default:
      instance = undefined;
  }

  return instance;
}

export { createElement };
