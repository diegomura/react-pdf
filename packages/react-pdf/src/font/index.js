import isUrl from 'is-url';
import standardFonts from './standard';
import { fetchFont } from '../utils/font';

let fonts = {};

const register = (src, { family, ...otherOptions }) => {
  fonts[family] = {
    src,
    loaded: false,
    data: null,
    ...otherOptions,
  };
};

const getRegisteredFonts = () => Object.keys(fonts);

const load = async (fontFamily, doc) => {
  const font = fonts[fontFamily];

  // We cache the font to avoid fetching it many time
  if (font && !font.data) {
    font.data = isUrl(font.src) ? await fetchFont(font.src) : font.src;
  }

  // If the font wasn't added to the document yet (aka. loaded), we do.
  // This prevents calling `registerFont` many times for the same font.
  // Fonts loaded state will be resetted after document is closed.
  if (font && !font.loaded) {
    font.loaded = true;
    doc.registerFont(fontFamily, font.data);
  }

  if (!font && !standardFonts.includes(fontFamily)) {
    throw new Error(
      `Font familiy not registered: ${fontFamily}. Please register it calling Font.register() method.`,
    );
  }
};

const reset = () => {
  for (const font in fonts) {
    if (fonts.hasOwnProperty(font)) {
      fonts[font].loaded = false;
    }
  }
};

const clear = () => {
  fonts = {};
};

export default {
  register,
  getRegisteredFonts,
  load,
  clear,
  reset,
};
