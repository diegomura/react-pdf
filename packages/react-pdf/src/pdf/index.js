/* global Blob */
import transformTree from './transformer';

const pdf = input => {
  const header = '%PDF-1.7\n';

  // Creates the footer of the PDF document
  function buildCrossReferenceTable(objects) {
    let offset = header.length - 1;

    const pointers = objects
      .map(object => {
        const pointer = `${('0000000000' + offset).slice(-10)} 00000 n`;
        offset += object.value.length;
        return pointer;
      })
      .join('\n');

    return 'xref\n' +
      `0 ${objects.length + 1}\n` +
      '0000000000 65535 f\n' +
      `${pointers}\n` +
      'trailer\n' +
      '<<\n' +
      `/Size ${objects.length}\n` +
      '/Root 1 0 R\n' +
      '>>' +
      'startxref\n' +
      `${offset}\n` +
      '%%EOF';
  }

  function buildDocument(input) {
    // Transforms the nodes trees into a more suitable data structure
    const objects = transformTree(input);

    // Builds the body and footer of the document
    const body = objects.map(object => object.value).join('\n\n');
    const footer = buildCrossReferenceTable(objects);

    return header + body + footer;
  }

  function toBlob() {
    return new Blob([buildDocument(input)], {
      type: 'application/pdf',
    });
  }

  function toBuffer() {
    return new Buffer(buildDocument(input));
  }

  return {
    toBuffer,
    toBlob,
  };
};

export default pdf;
