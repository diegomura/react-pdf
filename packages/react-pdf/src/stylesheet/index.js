import matchMedia from 'media-engine';
import transformStyles from './transformStyles';

const create = styles => styles;

const flatten = input => {
  if (!Array.isArray(input)) {
    input = [input];
  }

  const result = input.reduce((acc, style) => {
    if (style) {
      Object.keys(style).forEach(key => {
        if (style[key]) {
          acc[key] = style[key];
        }
      });
    }

    return acc;
  }, {});

  return result;
};

const resolveMediaQueries = (input, container) => {
  const result = Object.keys(input).reduce((acc, key) => {
    if (/@media/.test(key)) {
      return {
        ...acc,
        ...matchMedia({ [key]: input[key] }, container),
      };
    }

    return { ...acc, [key]: input[key] };
  }, {});

  return result;
};

const resolve = (styles, container) => {
  if (!styles) {
    return null;
  }

  styles = flatten(styles);
  styles = resolveMediaQueries(styles, container);

  return transformStyles(styles);
};

const absoluteFillObject = {
  position: 'absolute',
  top: 0,
  left: 0,
  bottom: 0,
  right: 0,
};

module.exports = {
  hairlineWidth: 1,
  create,
  resolve,
  flatten,
  absoluteFillObject,
};
