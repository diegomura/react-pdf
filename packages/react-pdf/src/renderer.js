'use strict';

import ReactFiberReconciler from 'react-reconciler';
import emptyObject from 'fbjs/lib/emptyObject';
import { createElement } from './elements';

const PDFRenderer = ReactFiberReconciler({
  appendInitialChild(parentInstance, child) {
    if (parentInstance.appendChild) {
      parentInstance.appendChild(child);
    } else {
      parentInstance.document = child;
    }
  },

  createInstance(type, props, internalInstanceHandle) {
    return createElement(type, props, internalInstanceHandle);
  },

  createTextInstance(text, rootContainerInstance, internalInstanceHandle) {
    return text;
  },

  finalizeInitialChildren(domElement, type, props) {
    return false;
  },

  getPublicInstance(instance) {
    return instance;
  },

  prepareForCommit() {
    // Noop
  },

  prepareUpdate(domElement, type, oldProps, newProps) {
    return true;
  },

  resetAfterCommit() {
    // Noop
  },

  resetTextContent(domElement) {
    // Noop
  },

  getRootHostContext() {
    return emptyObject;
  },

  getChildHostContext() {
    return emptyObject;
  },

  shouldSetTextContent(type, props) {
    return false;
  },

  now: () => {},

  useSyncScheduling: true,

  mutation: {
    appendChild(parentInstance, child) {
      if (parentInstance.appendChild) {
        parentInstance.appendChild(child);
      } else {
        parentInstance.document = child;
      }
    },

    appendChildToContainer(parentInstance, child) {
      if (parentInstance.appendChild) {
        parentInstance.appendChild(child);
      } else {
        parentInstance.document = child;
      }
    },

    insertBefore(parentInstance, child, beforeChild) {
      // noob
    },

    insertInContainerBefore(parentInstance, child, beforeChild) {
      // noob
    },

    removeChild(parentInstance, child) {
      parentInstance.removeChild(child);
    },

    removeChildFromContainer(parentInstance, child) {
      if (parentInstance.removeChild) {
        parentInstance.removeChild(child);
      }
    },

    commitTextUpdate(textInstance, oldText, newText) {
      textInstance = newText;
    },

    commitMount(instance, type, newProps) {
      // Noop
    },

    commitUpdate(instance, updatePayload, type, oldProps, newProps) {
      // noop
    },
  },
});

export default PDFRenderer;
