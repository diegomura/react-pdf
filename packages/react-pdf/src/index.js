'use strict';

import { PDFRenderer, createElement } from './renderer';
import StyleSheet from './Stylesheet';
import pdf from './pdf/index.js';

/* Component constants */
const View = 'VIEW';
const Text = 'TEXT';
const Page = 'PAGE';
const Document = 'DOCUMENT';

export {
  PDFRenderer,
  View,
  Text,
  Page,
  Document,
  StyleSheet,
  createElement,
  pdf,
};
