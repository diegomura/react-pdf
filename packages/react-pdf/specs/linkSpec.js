import React from 'react';
import { Document, Page, Text, Link } from '../src';
import MockDate from 'mockdate';
import render from './testRenderer';

const src = 'http://www.github.com/diegomura/react-pdf';

describe('<Link />', () => {
  beforeEach(() => {
    MockDate.set(1434319925275);
  });

  const matchSnapshot = (doc, done) =>
    render(<Document>{doc}</Document>).then(result => {
      expect(result).toMatchSnapshot();
      done();
    });

  test('Should render just a Link inside a Page', done => {
    matchSnapshot(
      <Page>
        <Link src={src}>link</Link>
      </Page>,
      done,
    );
  });

  test('Should render inline Link at the start of a Text', done => {
    matchSnapshot(
      <Page>
        <Text>
          <Link src={src}>link</Link>
          Lorem ipsum dolor sit amet
        </Text>
      </Page>,
      done,
    );
  });

  test('Should render inline Link in the middle of a Text', done => {
    matchSnapshot(
      <Page>
        <Text>
          Lorem ipsum dolor sit amet
          <Link src={src}>link</Link>
          Lorem ipsum dolor sit amet
        </Text>
      </Page>,
      done,
    );
  });

  test('Should render inline Link at the end of a Text', done => {
    matchSnapshot(
      <Page>
        <Text>
          Lorem ipsum dolor sit amet
          <Link src={src}>link</Link>
        </Text>
      </Page>,
      done,
    );
  });

  test('Should render just a Link using http', done => {
    const srcWithoutProtocol = 'www.github.com/diegomura/react-pdf';

    matchSnapshot(
      <Page>
        <Link src={srcWithoutProtocol}>link</Link>
      </Page>,
      done,
    );
  });

  test('Should render just a Link using http when given', done => {
    const srcWithProtocol = 'http://www.github.com/diegomura/react-pdf';

    matchSnapshot(
      <Page>
        <Link src={srcWithProtocol}>link</Link>
      </Page>,
      done,
    );
  });

  test('Should render just a Link using https when given', done => {
    const srcWithProtocol = 'https://www.github.com/diegomura/react-pdf';

    matchSnapshot(
      <Page>
        <Link src={srcWithProtocol}>link</Link>
      </Page>,
      done,
    );
  });

  test('Should render just a Link using ftp when given', done => {
    const srcWithProtocol = 'ftp://www.github.com/diegomura/react-pdf';

    matchSnapshot(
      <Page>
        <Link src={srcWithProtocol}>link</Link>
      </Page>,
      done,
    );
  });

  test('Should render just a Link using ftps when given', done => {
    const srcWithProtocol = 'ftps://www.github.com/diegomura/react-pdf';

    matchSnapshot(
      <Page>
        <Link src={srcWithProtocol}>link</Link>
      </Page>,
      done,
    );
  });

  test('Should render just a Link using mailto when given', done => {
    const srcWithProtocol = 'mailto://www.github.com/diegomura/react-pdf';

    matchSnapshot(
      <Page>
        <Link src={srcWithProtocol}>link</Link>
      </Page>,
      done,
    );
  });
});
