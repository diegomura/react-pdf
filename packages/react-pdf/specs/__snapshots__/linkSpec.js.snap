// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Should render inline Link at the end of a Text 1`] = `
"%PDF-1.3
%����
6 0 obj
<<
/Type /ExtGState
/ca 1
>>
endobj
8 0 obj
<<
/S /URI
/URI (http://www.github.com/diegomura/react-pdf)
>>
endobj
9 0 obj
<<
/Subtype /Link
/A 8 0 R
/Type /Annot
/Rect [217.062 825.24 244.062 841.89]
/Border [0 0 0]
>>
endobj
10 0 obj
<<
/Type /ExtGState
/CA 1
>>
endobj
5 0 obj
<<
/Type /Page
/Parent 1 0 R
/MediaBox [0 0 595.28 841.89]
/Contents 3 0 R
/Resources 4 0 R
/Annots [9 0 R]
>>
endobj
4 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <<
/Gs1 6 0 R
/Gs2 10 0 R
>>
/Font <<
/F1 7 0 R
>>
>>
endobj
11 0 obj
<<
/Producer (react-pdf)
/Creator (react-pdf)
/CreationDate (D:20150614221205Z)
>>
endobj
7 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
2 0 obj
<<
/Type /Catalog
/Pages 1 0 R
>>
endobj
1 0 obj
<<
/Type /Pages
/Count 1
/Kids [5 0 R]
>>
endobj
3 0 obj
<<
/Length 198
/Filter /FlateDecode
>>
stream
x���M
�0��s���i&���E�\\\\��či+����7��E	����7C(��/�&��j8��Iy޻r�%��ָ#YC�k<C��&��;YE\\"fF[C0#�m��v\\\\�!Jrl���]sŝ����5��C��(7h�0�d{��%���=����P-AxEFHVH��kPZ?��}��[���e~ޘK�>�w\`�[�
endstream
endobj
xref
0 12
0000000000 65535 f 
0000000794 00000 n 
0000000745 00000 n 
0000000851 00000 n 
0000000420 00000 n 
0000000294 00000 n 
0000000015 00000 n 
0000000648 00000 n 
0000000059 00000 n 
0000000137 00000 n 
0000000249 00000 n 
0000000549 00000 n 
trailer
<<
/Size 12
/Root 2 0 R
/Info 11 0 R
>>
startxref
1121
%%EOF
"
`;

exports[`<Link /> Should render inline Link at the start of a Text 1`] = `
"%PDF-1.3
%����
6 0 obj
<<
/Type /ExtGState
/ca 1
>>
endobj
7 0 obj
<<
/S /URI
/URI (http://www.github.com/diegomura/react-pdf)
>>
endobj
8 0 obj
<<
/Subtype /Link
/A 7 0 R
/Type /Annot
/Rect [0 825.24 27 841.89]
/Border [0 0 0]
>>
endobj
9 0 obj
<<
/Type /ExtGState
/CA 1
>>
endobj
5 0 obj
<<
/Type /Page
/Parent 1 0 R
/MediaBox [0 0 595.28 841.89]
/Contents 3 0 R
/Resources 4 0 R
/Annots [8 0 R]
>>
endobj
4 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <<
/Gs1 6 0 R
/Gs2 9 0 R
>>
/Font <<
/F1 10 0 R
>>
>>
endobj
11 0 obj
<<
/Producer (react-pdf)
/Creator (react-pdf)
/CreationDate (D:20150614221205Z)
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
2 0 obj
<<
/Type /Catalog
/Pages 1 0 R
>>
endobj
1 0 obj
<<
/Type /Pages
/Count 1
/Kids [5 0 R]
>>
endobj
3 0 obj
<<
/Length 191
/Filter /FlateDecode
>>
stream
x���A�@���+��;���Bt�J��{�.�A\\"��o5	F10�}��
?3�+RD1��C�*WW�]��Zw�0k	/��O#��$�| �t��#�^%�����>��j��A�e13��
!Eh+8.�q�%��(Nh�����;���WF��B
��0s��W�{�k�Q^#.X5*���Xl
endstream
endobj
xref
0 12
0000000000 65535 f 
0000000783 00000 n 
0000000734 00000 n 
0000000840 00000 n 
0000000408 00000 n 
0000000282 00000 n 
0000000015 00000 n 
0000000059 00000 n 
0000000137 00000 n 
0000000238 00000 n 
0000000636 00000 n 
0000000537 00000 n 
trailer
<<
/Size 12
/Root 2 0 R
/Info 11 0 R
>>
startxref
1103
%%EOF
"
`;

exports[`<Link /> Should render inline Link in the middle of a Text 1`] = `
"%PDF-1.3
%����
6 0 obj
<<
/Type /ExtGState
/ca 1
>>
endobj
8 0 obj
<<
/S /URI
/URI (http://www.github.com/diegomura/react-pdf)
>>
endobj
9 0 obj
<<
/Subtype /Link
/A 8 0 R
/Type /Annot
/Rect [217.062 825.24 244.062 841.89]
/Border [0 0 0]
>>
endobj
10 0 obj
<<
/Type /ExtGState
/CA 1
>>
endobj
5 0 obj
<<
/Type /Page
/Parent 1 0 R
/MediaBox [0 0 595.28 841.89]
/Contents 3 0 R
/Resources 4 0 R
/Annots [9 0 R]
>>
endobj
4 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <<
/Gs1 6 0 R
/Gs2 10 0 R
>>
/Font <<
/F1 7 0 R
>>
>>
endobj
11 0 obj
<<
/Producer (react-pdf)
/Creator (react-pdf)
/CreationDate (D:20150614221205Z)
>>
endobj
7 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
2 0 obj
<<
/Type /Catalog
/Pages 1 0 R
>>
endobj
1 0 obj
<<
/Type /Pages
/Count 1
/Kids [5 0 R]
>>
endobj
3 0 obj
<<
/Length 208
/Filter /FlateDecode
>>
stream
x���M
�0��s���m&���E�\\\\��či+����7��?�	���/ɛ	�pc@n
ya���,����-Vi���b�=��ք��1��hl:��e�E�h*�g��)a=R�K-9�\\\\
���C}[+.���Ӝ��F�s�������l�>=d�I�Y�h����=�)�8�
�R��2��s酫7₷?��ū]���-��
endstream
endobj
xref
0 12
0000000000 65535 f 
0000000794 00000 n 
0000000745 00000 n 
0000000851 00000 n 
0000000420 00000 n 
0000000294 00000 n 
0000000015 00000 n 
0000000648 00000 n 
0000000059 00000 n 
0000000137 00000 n 
0000000249 00000 n 
0000000549 00000 n 
trailer
<<
/Size 12
/Root 2 0 R
/Info 11 0 R
>>
startxref
1131
%%EOF
"
`;

exports[`<Link /> Should render just a Link inside a Page 1`] = `
"%PDF-1.3
%����
6 0 obj
<<
/Type /ExtGState
/ca 1
>>
endobj
7 0 obj
<<
/S /URI
/URI (http://www.github.com/diegomura/react-pdf)
>>
endobj
8 0 obj
<<
/Subtype /Link
/A 7 0 R
/Type /Annot
/Rect [0 825.24 27 841.89]
/Border [0 0 0]
>>
endobj
9 0 obj
<<
/Type /ExtGState
/CA 1
>>
endobj
5 0 obj
<<
/Type /Page
/Parent 1 0 R
/MediaBox [0 0 595.28 841.89]
/Contents 3 0 R
/Resources 4 0 R
/Annots [8 0 R]
>>
endobj
4 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <<
/Gs1 6 0 R
/Gs2 9 0 R
>>
/Font <<
/F1 10 0 R
>>
>>
endobj
11 0 obj
<<
/Producer (react-pdf)
/Creator (react-pdf)
/CreationDate (D:20150614221205Z)
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
2 0 obj
<<
/Type /Catalog
/Pages 1 0 R
>>
endobj
1 0 obj
<<
/Type /Pages
/Count 1
/Kids [5 0 R]
>>
endobj
3 0 obj
<<
/Length 146
/Filter /FlateDecode
>>
stream
x�mͱ
1�=O���^lA��A�M\\\\,U+}}S�� �~>�623�e椌Ő\`��u|�C<�=���=f�[�#W��E�E�%�Nq�	�E=��~����5�ƭQ�}�%$��
�%������~/}�L,<
endstream
endobj
xref
0 12
0000000000 65535 f 
0000000783 00000 n 
0000000734 00000 n 
0000000840 00000 n 
0000000408 00000 n 
0000000282 00000 n 
0000000015 00000 n 
0000000059 00000 n 
0000000137 00000 n 
0000000238 00000 n 
0000000636 00000 n 
0000000537 00000 n 
trailer
<<
/Size 12
/Root 2 0 R
/Info 11 0 R
>>
startxref
1058
%%EOF
"
`;

exports[`<Link /> Should render just a Link using ftp when given 1`] = `
"%PDF-1.3
%����
6 0 obj
<<
/Type /ExtGState
/ca 1
>>
endobj
7 0 obj
<<
/S /URI
/URI (ftp://www.github.com/diegomura/react-pdf)
>>
endobj
8 0 obj
<<
/Subtype /Link
/A 7 0 R
/Type /Annot
/Rect [0 825.24 27 841.89]
/Border [0 0 0]
>>
endobj
9 0 obj
<<
/Type /ExtGState
/CA 1
>>
endobj
5 0 obj
<<
/Type /Page
/Parent 1 0 R
/MediaBox [0 0 595.28 841.89]
/Contents 3 0 R
/Resources 4 0 R
/Annots [8 0 R]
>>
endobj
4 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <<
/Gs1 6 0 R
/Gs2 9 0 R
>>
/Font <<
/F1 10 0 R
>>
>>
endobj
11 0 obj
<<
/Producer (react-pdf)
/Creator (react-pdf)
/CreationDate (D:20150614221205Z)
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
2 0 obj
<<
/Type /Catalog
/Pages 1 0 R
>>
endobj
1 0 obj
<<
/Type /Pages
/Count 1
/Kids [5 0 R]
>>
endobj
3 0 obj
<<
/Length 146
/Filter /FlateDecode
>>
stream
x�mͱ
1�=O���^lA��A�M\\\\,U+}}S�� �~>�623�e椌Ő\`��u|�C<�=���=f�[�#W��E�E�%�Nq�	�E=��~����5�ƭQ�}�%$��
�%������~/}�L,<
endstream
endobj
xref
0 12
0000000000 65535 f 
0000000782 00000 n 
0000000733 00000 n 
0000000839 00000 n 
0000000407 00000 n 
0000000281 00000 n 
0000000015 00000 n 
0000000059 00000 n 
0000000136 00000 n 
0000000237 00000 n 
0000000635 00000 n 
0000000536 00000 n 
trailer
<<
/Size 12
/Root 2 0 R
/Info 11 0 R
>>
startxref
1057
%%EOF
"
`;

exports[`<Link /> Should render just a Link using ftps when given 1`] = `
"%PDF-1.3
%����
6 0 obj
<<
/Type /ExtGState
/ca 1
>>
endobj
7 0 obj
<<
/S /URI
/URI (ftps://www.github.com/diegomura/react-pdf)
>>
endobj
8 0 obj
<<
/Subtype /Link
/A 7 0 R
/Type /Annot
/Rect [0 825.24 27 841.89]
/Border [0 0 0]
>>
endobj
9 0 obj
<<
/Type /ExtGState
/CA 1
>>
endobj
5 0 obj
<<
/Type /Page
/Parent 1 0 R
/MediaBox [0 0 595.28 841.89]
/Contents 3 0 R
/Resources 4 0 R
/Annots [8 0 R]
>>
endobj
4 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <<
/Gs1 6 0 R
/Gs2 9 0 R
>>
/Font <<
/F1 10 0 R
>>
>>
endobj
11 0 obj
<<
/Producer (react-pdf)
/Creator (react-pdf)
/CreationDate (D:20150614221205Z)
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
2 0 obj
<<
/Type /Catalog
/Pages 1 0 R
>>
endobj
1 0 obj
<<
/Type /Pages
/Count 1
/Kids [5 0 R]
>>
endobj
3 0 obj
<<
/Length 146
/Filter /FlateDecode
>>
stream
x�mͱ
1�=O���^lA��A�M\\\\,U+}}S�� �~>�623�e椌Ő\`��u|�C<�=���=f�[�#W��E�E�%�Nq�	�E=��~����5�ƭQ�}�%$��
�%������~/}�L,<
endstream
endobj
xref
0 12
0000000000 65535 f 
0000000783 00000 n 
0000000734 00000 n 
0000000840 00000 n 
0000000408 00000 n 
0000000282 00000 n 
0000000015 00000 n 
0000000059 00000 n 
0000000137 00000 n 
0000000238 00000 n 
0000000636 00000 n 
0000000537 00000 n 
trailer
<<
/Size 12
/Root 2 0 R
/Info 11 0 R
>>
startxref
1058
%%EOF
"
`;

exports[`<Link /> Should render just a Link using http 1`] = `
"%PDF-1.3
%����
6 0 obj
<<
/Type /ExtGState
/ca 1
>>
endobj
7 0 obj
<<
/S /URI
/URI (http://www.github.com/diegomura/react-pdf)
>>
endobj
8 0 obj
<<
/Subtype /Link
/A 7 0 R
/Type /Annot
/Rect [0 825.24 27 841.89]
/Border [0 0 0]
>>
endobj
9 0 obj
<<
/Type /ExtGState
/CA 1
>>
endobj
5 0 obj
<<
/Type /Page
/Parent 1 0 R
/MediaBox [0 0 595.28 841.89]
/Contents 3 0 R
/Resources 4 0 R
/Annots [8 0 R]
>>
endobj
4 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <<
/Gs1 6 0 R
/Gs2 9 0 R
>>
/Font <<
/F1 10 0 R
>>
>>
endobj
11 0 obj
<<
/Producer (react-pdf)
/Creator (react-pdf)
/CreationDate (D:20150614221205Z)
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
2 0 obj
<<
/Type /Catalog
/Pages 1 0 R
>>
endobj
1 0 obj
<<
/Type /Pages
/Count 1
/Kids [5 0 R]
>>
endobj
3 0 obj
<<
/Length 146
/Filter /FlateDecode
>>
stream
x�mͱ
1�=O���^lA��A�M\\\\,U+}}S�� �~>�623�e椌Ő\`��u|�C<�=���=f�[�#W��E�E�%�Nq�	�E=��~����5�ƭQ�}�%$��
�%������~/}�L,<
endstream
endobj
xref
0 12
0000000000 65535 f 
0000000783 00000 n 
0000000734 00000 n 
0000000840 00000 n 
0000000408 00000 n 
0000000282 00000 n 
0000000015 00000 n 
0000000059 00000 n 
0000000137 00000 n 
0000000238 00000 n 
0000000636 00000 n 
0000000537 00000 n 
trailer
<<
/Size 12
/Root 2 0 R
/Info 11 0 R
>>
startxref
1058
%%EOF
"
`;

exports[`<Link /> Should render just a Link using http when given 1`] = `
"%PDF-1.3
%����
6 0 obj
<<
/Type /ExtGState
/ca 1
>>
endobj
7 0 obj
<<
/S /URI
/URI (http://www.github.com/diegomura/react-pdf)
>>
endobj
8 0 obj
<<
/Subtype /Link
/A 7 0 R
/Type /Annot
/Rect [0 825.24 27 841.89]
/Border [0 0 0]
>>
endobj
9 0 obj
<<
/Type /ExtGState
/CA 1
>>
endobj
5 0 obj
<<
/Type /Page
/Parent 1 0 R
/MediaBox [0 0 595.28 841.89]
/Contents 3 0 R
/Resources 4 0 R
/Annots [8 0 R]
>>
endobj
4 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <<
/Gs1 6 0 R
/Gs2 9 0 R
>>
/Font <<
/F1 10 0 R
>>
>>
endobj
11 0 obj
<<
/Producer (react-pdf)
/Creator (react-pdf)
/CreationDate (D:20150614221205Z)
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
2 0 obj
<<
/Type /Catalog
/Pages 1 0 R
>>
endobj
1 0 obj
<<
/Type /Pages
/Count 1
/Kids [5 0 R]
>>
endobj
3 0 obj
<<
/Length 146
/Filter /FlateDecode
>>
stream
x�mͱ
1�=O���^lA��A�M\\\\,U+}}S�� �~>�623�e椌Ő\`��u|�C<�=���=f�[�#W��E�E�%�Nq�	�E=��~����5�ƭQ�}�%$��
�%������~/}�L,<
endstream
endobj
xref
0 12
0000000000 65535 f 
0000000783 00000 n 
0000000734 00000 n 
0000000840 00000 n 
0000000408 00000 n 
0000000282 00000 n 
0000000015 00000 n 
0000000059 00000 n 
0000000137 00000 n 
0000000238 00000 n 
0000000636 00000 n 
0000000537 00000 n 
trailer
<<
/Size 12
/Root 2 0 R
/Info 11 0 R
>>
startxref
1058
%%EOF
"
`;

exports[`<Link /> Should render just a Link using https when given 1`] = `
"%PDF-1.3
%����
6 0 obj
<<
/Type /ExtGState
/ca 1
>>
endobj
7 0 obj
<<
/S /URI
/URI (https://www.github.com/diegomura/react-pdf)
>>
endobj
8 0 obj
<<
/Subtype /Link
/A 7 0 R
/Type /Annot
/Rect [0 825.24 27 841.89]
/Border [0 0 0]
>>
endobj
9 0 obj
<<
/Type /ExtGState
/CA 1
>>
endobj
5 0 obj
<<
/Type /Page
/Parent 1 0 R
/MediaBox [0 0 595.28 841.89]
/Contents 3 0 R
/Resources 4 0 R
/Annots [8 0 R]
>>
endobj
4 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <<
/Gs1 6 0 R
/Gs2 9 0 R
>>
/Font <<
/F1 10 0 R
>>
>>
endobj
11 0 obj
<<
/Producer (react-pdf)
/Creator (react-pdf)
/CreationDate (D:20150614221205Z)
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
2 0 obj
<<
/Type /Catalog
/Pages 1 0 R
>>
endobj
1 0 obj
<<
/Type /Pages
/Count 1
/Kids [5 0 R]
>>
endobj
3 0 obj
<<
/Length 146
/Filter /FlateDecode
>>
stream
x�mͱ
1�=O���^lA��A�M\\\\,U+}}S�� �~>�623�e椌Ő\`��u|�C<�=���=f�[�#W��E�E�%�Nq�	�E=��~����5�ƭQ�}�%$��
�%������~/}�L,<
endstream
endobj
xref
0 12
0000000000 65535 f 
0000000784 00000 n 
0000000735 00000 n 
0000000841 00000 n 
0000000409 00000 n 
0000000283 00000 n 
0000000015 00000 n 
0000000059 00000 n 
0000000138 00000 n 
0000000239 00000 n 
0000000637 00000 n 
0000000538 00000 n 
trailer
<<
/Size 12
/Root 2 0 R
/Info 11 0 R
>>
startxref
1059
%%EOF
"
`;

exports[`<Link /> Should render just a Link using mailto when given 1`] = `
"%PDF-1.3
%����
6 0 obj
<<
/Type /ExtGState
/ca 1
>>
endobj
7 0 obj
<<
/S /URI
/URI (mailto://www.github.com/diegomura/react-pdf)
>>
endobj
8 0 obj
<<
/Subtype /Link
/A 7 0 R
/Type /Annot
/Rect [0 825.24 27 841.89]
/Border [0 0 0]
>>
endobj
9 0 obj
<<
/Type /ExtGState
/CA 1
>>
endobj
5 0 obj
<<
/Type /Page
/Parent 1 0 R
/MediaBox [0 0 595.28 841.89]
/Contents 3 0 R
/Resources 4 0 R
/Annots [8 0 R]
>>
endobj
4 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/ExtGState <<
/Gs1 6 0 R
/Gs2 9 0 R
>>
/Font <<
/F1 10 0 R
>>
>>
endobj
11 0 obj
<<
/Producer (react-pdf)
/Creator (react-pdf)
/CreationDate (D:20150614221205Z)
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
2 0 obj
<<
/Type /Catalog
/Pages 1 0 R
>>
endobj
1 0 obj
<<
/Type /Pages
/Count 1
/Kids [5 0 R]
>>
endobj
3 0 obj
<<
/Length 146
/Filter /FlateDecode
>>
stream
x�mͱ
1�=O���^lA��A�M\\\\,U+}}S�� �~>�623�e椌Ő\`��u|�C<�=���=f�[�#W��E�E�%�Nq�	�E=��~����5�ƭQ�}�%$��
�%������~/}�L,<
endstream
endobj
xref
0 12
0000000000 65535 f 
0000000785 00000 n 
0000000736 00000 n 
0000000842 00000 n 
0000000410 00000 n 
0000000284 00000 n 
0000000015 00000 n 
0000000059 00000 n 
0000000139 00000 n 
0000000240 00000 n 
0000000638 00000 n 
0000000539 00000 n 
trailer
<<
/Size 12
/Root 2 0 R
/Info 11 0 R
>>
startxref
1060
%%EOF
"
`;
